client/packet_test.go

```diff
@@ -0,0 +1,144 @@
+package client
+
+import (
+	"net"
+	"reflect"
+	"testing"
+)
+
+var mac = net.HardwareAddr([]byte{0xb8, 0xae, 0xed, 0x7a, 0x10, 0x66})
+
+func TestNewSolicitOptions(t *testing.T) {
+	options, err := newSolicitOptions(mac)
+	if err != nil {
+		t.Fatalf("error in newSolicitOptions: %v\n", err)
+	}
+	expected := Options(map[OptionCode][][]byte{
+		OptionIANA:        [][]byte{[]byte{0x72, 0x6f, 0x6f, 0x74, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00}},
```

Comment:
You can omit the `[]byte` part of each element and just do `[][]byte{{0xff}}` to simplify things a bit.